frontend/src/metabase-lib/parameters/utils/template-tags.unit.spec.js

```diff
@@ -161,6 +168,38 @@ describe("parameters/utils/cards", () => {
         parametersWithFieldFilterOperatorTypes,
       );
     });
+
+    it("should exclude tags that are not parameters", () => {
+      const tags = [
+        createMockTemplateTag({
+          type: "string",
+          id: "1",
+          name: "a",
+          "display-name": "A",
+        }),
+        createMockTemplateTag({
+          type: "card",
+          id: "2",
+          "card-id": 123,
+        }),
+        createMockTemplateTag({
+          type: "snippet",
+          id: "3",
+          "snippet-id": 1,
+          "snippet-name": "C",
+        }),
+      ];
+      expect(getTemplateTagParameters(tags)).toEqual([
```

Comment:
should we write this test using negation? I think it's easier to understand what we want to test here if we write this test as  "template tag does not contains snippet" instead of using equal here.

Reply:
Writing this test with negation is less safe I think. There are many changes I could make to code that would break the functionality but keep the test passing if it were written with negation. The same is not true with the way this test is currently written.

For example, if I wrote this test with negation in clojure, it might be:
```
(is (empty (set/intersection #{:snippet :card) (map :type (get-template-tag-parameters tags))
```
If the way `:type` is represented on a `template-tag` changes, e.g. `{:type :snippet}` changes to `{:type "snippet"}` then this test would still pass even though `get-template-tag-parameters` can be completely broken.

As for understanding, I think the test description is enough to understand pretty quickly what is going on.